x509/x509.go

```diff
@@ -1272,6 +1314,41 @@ func parsePublicKey(algo PublicKeyAlgorithm, keyData *publicKeyInfo, nfe *NonFat
 			return nil, errors.New("x509: RSA public exponent is not a positive number")
 		}
 
+		pub := &rsa.PublicKey{
+			E: p.E,
+			N: p.N,
+		}
+		return pub, nil
+	case RSAESOAEP:
+		p := new(pkcs1PublicKey)
+		rest, err := asn1.Unmarshal(asn1Data, p)
+		if err != nil {
+			var laxErr error
+			rest, laxErr = asn1.UnmarshalWithParams(asn1Data, p, "lax")
+			if laxErr != nil {
+				return nil, laxErr
+			}
+			nfe.AddError(err)
+		}
+		if len(rest) != 0 {
+			return nil, errors.New("x509: trailing data after RSA public key")
+		}
+		if p.N.Sign() <= 0 {
+			nfe.AddError(errors.New("x509: RSA modulus is not a positive number"))
+		}
+		if p.E <= 0 {
+			return nil, errors.New("x509: RSA public exponent is not a positive number")
+		}
```

Comment:
Could do with a comment here pointing out that we're checking the `Parameters` format but otherwise ignoring/dropping it.

Reply:
By the way, is the public key usable without the parameter values?  (I don't know anything about RSAES-OAEP)  Should there be a `TODO(someone): export parameter values` or a `TODO(someone): replace this functionality with whatever upstream `crypto/x509` ends up supporting` ?

Reply:
Technically no it shouldn't be. You need the parameters for the RSAES-OAEP encryption scheme. The key itself is still just rsa and can be used as such but it really shouldn't be (in my opinion it would be best to modify either the PublicKeyAlgorithm in the Certificate struct to hold the parameters relevant to the scheme, but that might be considered an aggressive change). I'm still not sure what upstream will do so I've left off on making any such changes. I've added a TODO.

Reply:
Is there an upstream issue we can track (and mention here)?

Reply:
I've mostly been talking with upstream internally, there is an open issue and I've linked it.